crates/transaction-pool/src/pool/parked.rs

```diff
@@ -91,7 +91,7 @@ impl<T: ParkedOrd> ParkedPool<T> {
         self.by_id
             .range((sender.start_bound(), Unbounded))
             .take_while(move |(other, _)| sender == other.sender)
-            .map(|(_, tx)| *tx.transaction.id())
+            .map(|(tx_id, _)| *tx_id)
```

Comment:
@mattsse here maybe this is not needed but why storing only the transaction ID and not the complete transaction so that instead of a vector of transaction IDs we have a vector with full transaction information and we can extract the ID for each anyway wdt?

Reply:
can you elaborate on that a bit?
do you mean the return type?

the only purpose of this function is to get the ids when we truncate the pool and we only really need the ids here

Reply:
@mattsse Yes the return type, I mean that for the moment with this method we only return a vector of `TransactionId` (which is fine for the moment because on the only place where this function is used, we only need the `TransactionId`). 

But why not have more information available by returning a vector of `Arc<ValidPoolTransaction<T::Transaction>>`. In this way, we would always have access to the `TransactionId` which is inside but also to the exhaustive information of the transaction (hash, timestamp and all the others). What do you think? Too much storage cost perhaps? No use at the moment?